FWCore/Concurrency/interface/WaitingTaskHolder.h

```diff
@@ -0,0 +1,84 @@
+#ifndef FWCore_Concurrency_WaitingTaskHolder_h
+#define FWCore_Concurrency_WaitingTaskHolder_h
+// -*- C++ -*-
+//
+// Package:     FWCore/Concurrency
+// Class  :     WaitingTaskHolder
+// 
+/**\class WaitingTaskHolder WaitingTaskHolder.h "WaitingTaskHolder.h"
+
+ Description: [one line class summary]
+
+ Usage:
+    <usage>
+
+*/
+//
+// Original Author:  FWCore
+//         Created:  Fri, 18 Nov 2016 20:30:42 GMT
+//
+
+// system include files
+#include <cassert>
+
+// user include files
+#include "FWCore/Concurrency/interface/WaitingTask.h"
+
+// forward declarations
+
+namespace edm {
+  class WaitingTaskHolder
+  {
+    
+  public:
+    WaitingTaskHolder():
+    m_task(nullptr) {}
+    
```

Comment:
Somewhat a matter of taste, but initializing `m_task` to `nullptr` in-class allows you to just specify:

    WaitingTaskHolder() = default;

Of course, doing so will not save you much maintenance for this class since there's only one data member.  But extending the class in the future is then easier.